File: refactor_consultation.js
const fs = require('fs');

const path = './frontend/src/pages/ConsultationFormPage.tsx';
let content = fs.readFileSync(path, 'utf8');

// 1. Import Section
if (!content.includes('import { AppCard, AppButton, AppTextField, PageHeader, Section }')) {
  content = content.replace(
    /import \{ AppCard, AppButton, AppTextField, PageHeader \} from '\.\.\/ui';/g,
    "import { AppCard, AppButton, AppTextField, PageHeader, Section } from '../ui';"
  );
  // Just in case it wasn't there
  if (!content.includes('../ui')) {
     content = content.replace(
       /from '@mui\/material';\n/,
       "from '@mui/material';\nimport { AppCard, AppButton, AppTextField, PageHeader, Section } from '../ui';\n"
     );
  }
}

// 2. Replace Accordion with Section
// Pattern:
// <Accordion(?: defaultExpanded)?>
//   <AccordionSummary expandIcon={<ExpandMore />}>
//     <Typography variant="h6">(.*?)</Typography>
//   </AccordionSummary>
//   <AccordionDetails sx={{ p: { xs: 2, sm: 3 } }}>
//     ... content ...
//   </AccordionDetails>
// </Accordion>

const regex = /<Accordion( defaultExpanded)?>[\s\S]*?<Typography variant="h6">(.*?)<\/Typography>[\s\S]*?<AccordionDetails[^>]*>([\s\S]*?)<\/AccordionDetails>\s*<\/Accordion>/g;

content = content.replace(regex, (match, defaultExpanded, title, innerContent) => {
  const expandedProp = defaultExpanded ? ' defaultExpanded' : '';
  return `<Section title="${title}"${expandedProp}>\n${innerContent}\n</Section>`;
});

// 3. Let's do a basic replacement of TextField with AppTextField
content = content.replace(/<TextField/g, '<AppTextField');

// 4. Clean up unused imports like Accordion, AccordionSummary, AccordionDetails
content = content.replace(/\s*Accordion,\n/g, '\n');
content = content.replace(/\s*AccordionSummary,\n/g, '\n');
content = content.replace(/\s*AccordionDetails,\n/g, '\n');

fs.writeFileSync(path, content, 'utf8');
console.log('Refactoring complete!');

File: refactor_consultation.py
import os
import re

file_path = './frontend/src/pages/ConsultationFormPage.tsx'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 2. Replace Accordion with Section
# Using a simpler regex that matches any character lazy
pattern = r'<Accordion( defaultExpanded)?>.*?<Typography variant="h6">(.*?)<\/Typography>.*?<AccordionDetails[^>]*>(.*?)<\/AccordionDetails>\s*<\/Accordion>'

def replace_accordion(match):
    expanded = match.group(1) or ""
    title = match.group(2)
    inner = match.group(3)
    return f'<Section title="{title}"{expanded}>{inner}</Section>'

content = re.sub(pattern, replace_accordion, content, flags=re.DOTALL)

# Clean up imports
content = re.sub(r'\s*Accordion,', '', content)
content = re.sub(r'\s*AccordionSummary,', '', content)
content = re.sub(r'\s*AccordionDetails,', '', content)
content = re.sub(r'\s*TextField,', '', content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('Refactoring complete 2!')
